src/Common/LRUFileCache.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include <Common/IFileCachePriority.h>
+
+namespace DB
+{
+
+class LRUFileCache : public IFileCachePriority
+{
+public:
+    using LRUQueue = std::list<FileCacheRecord>;
+    using LRUQueueIterator = typename LRUQueue::iterator;
+    class WriteableIterator;
+
+    class ReadableIterator : public IIterator
```

Comment:
When an iterator uses a read operation, it should be prohibited from being used for a write operation. The iterator that is also used for a write operation should also be used for a read operation. I do this to prevent the iterator from being mixed when implementing functions. 
It may be possible to put the read and write functions in a class, but if someone is not clear about the relevant code to modify it, it may cause debugging difficulties.

Reply:
Why not just return const iterator in the method which need only read operations, and non const iterator when we need write operations? It will do the same thing as you described (methods for read only operations need to be marked const).

Reply:
Yes, looks like a good idea, I didn't think of it, I'll modify it like this. :)